src/main/java/seedu/address/ui/PersonDetailsWindow.java

```diff
@@ -31,124 +35,228 @@ public class PersonDetailsWindow extends UiPart<Stage> {
     private TextField addressField;
 
     /**
-     * Creates a new PersonDetailsWindow.
+     * Creates a new {@code PersonDetailsWindow} with the given {@code Stage} and {@code Logic}.
      *
-     * @param root Stage to use as the root of the PersonDetailsWindow.
+     * @param root  Stage to use as the root of the PersonDetailsWindow.
+     * @param logic The logic component to execute commands.
      */
     public PersonDetailsWindow(Stage root, Logic logic) {
         super(FXML, root);
         this.logic = logic;
-        getRoot().setAlwaysOnTop(true);
+        initializeKeyHandlers();
     }
 
     /**
-     * Create a new PersonDetailsWindow。
+     * Creates a new {@code PersonDetailsWindow} with the given {@code Logic}.
+     * A new {@code Stage} is created as the root.
      *
-     * @param logic to execute instructions
+     * @param logic The logic component to execute commands.
      */
     public PersonDetailsWindow(Logic logic) {
         this(new Stage(), logic);
     }
 
     /**
-     * Creates a new PersonDetailsWindow.
+     * Initializes keyboard event handlers for navigation and actions.
+     */
+    private void initializeKeyHandlers() {
+        // Add key handlers to each text field
+        addKeyHandlers(nameField);
+        addKeyHandlers(phoneField);
+        addKeyHandlers(emailField);
+        addKeyHandlers(addressField);
+    }
+
+    /**
+     * Adds keyboard event handlers to a specific {@code TextField}.
+     *
+     * @param textField The TextField to which the handlers are added.
+     */
+    private void addKeyHandlers(TextField textField) {
+        textField.addEventFilter(KeyEvent.KEY_PRESSED, event -> {
+            if (event.getCode() == KeyCode.DOWN) {
+                moveFocusToNext(textField);
+                event.consume();
+            } else if (event.getCode() == KeyCode.UP) {
+                moveFocusToPrevious(textField);
+                event.consume();
+            } else if (event.getCode() == KeyCode.ESCAPE) {
+                handleCancel();
+                event.consume();
+            } else if (event.getCode() == KeyCode.ENTER) {
+                handleSave();
+                event.consume();
+            }
+        });
+    }
+
+    /**
+     * Moves focus to the next {@code TextField} in the form.
+     *
+     * @param current The current TextField.
+     */
+    private void moveFocusToNext(TextField current) {
+        if (current == nameField) {
+            phoneField.requestFocus();
+        } else if (current == phoneField) {
+            emailField.requestFocus();
+        } else if (current == emailField) {
+            addressField.requestFocus();
+        } else if (current == addressField) {
+            nameField.requestFocus();
+        }
+    }
+
+    /**
+     * Moves focus to the previous {@code TextField} in the form.
+     *
+     * @param current The current TextField.
+     */
+    private void moveFocusToPrevious(TextField current) {
```

Comment:
Adequate demonstration of SLAP as the scope of the method is clearly contained within the level of abstraction.